src/lib/components/identity/identity-details.svelte

```diff
@@ -29,6 +30,15 @@
         }
         revoking = false
     }
+
+    async function handleRoleChange(role: UserRoles): Promise<void> {
+        identity.role = role
```

Comment:
Maybe add a check if the role is even changed. This is the case when the user selects the same role as before in the dropdown.